### DISCUSSION TITLE
umask doesn't seem to be respected for the devcontainer process/instance

### DISCUSSION THREAD

**Discussion by dss010101** [devcontainers/community#73] [Q&A]
Im running an amazon al2023 image in a remote devcontainer.   I have the following in the Dockerfile with hopes that files created within the container on mounted shares are read writeable by the dev team members. 

```RUN mkdir -p /home/$USER_NAME \
    && echo "umask 0000" >> /home/$USER_NAME/.bashrc \
    && chown -R $USER_NAME /home/$USER_NAME
....
....
RUN mkdir -p $DATA_HOME
RUN chown -R $USER_NAME $DATA_HOME
RUN chmod -R g+rwx $DATA_HOME
RUN setfacl -Rdm g::rwx $DATA_HOME
```
$DATA_HOME above will be a mounted volume.

Within the VSCode's devcontainer, if i open a terminal and touch a file on the share, the permissions are fine - they are read writeable by the group.  However, if i use visual studio code to create a file (File->New File), the file is r/w by only me.   I was looking through the devcontainer build/start logs and noticed the following:

```
Start: Run in container: cat '/home/xxxxx/.vscode-server-insiders/data/Machine/.connection-token-8bac42dd1ec2bb223e847a16437b196c040680e5-insider' 2>/dev/null || (**umask 377** && echo 'd4e6cea0-208e-4030-ad16-cd4b3fcb4a42' >
```
Not sure if that is the mask that is used for the devcontainer process/instance...but if it is, how would i change that to 0002?
Or is there another solution for solving this issue?

It is quite a productivity problem when we constantly have to reach out to team members of admins to change file permissions 

Thanks much

**Comment by samruddhikhandale** [comment]
Hi 👋 

To me, it sounds more like an issue with the Dev Containers Extension or VS Code. @dss010101 Do you mind opening an issue in https://github.com/microsoft/vscode-remote-release/issues? Thanks!

**Reply by dss010101** [reply]
Ok, i will.  thank you
